addons/analytic/models/analytic_plan.py

```diff
@@ -290,6 +290,31 @@ def _sync_plan_column(self):
                     'index': True,
                 })
 
+    def write(self, vals):
```

Comment:
I think it would be easier to write the logic like this, to also avoid having potential unbound variables etc (even if it's not the case, it's not obvious)
```python
def write(self, vals):
    new_parent = self.env['account.analytic.plan'].browse(vals.get('parent_id'))
    plan2previous_parent = {plan: plan.parent_id for plan in self if plan.parent_id}
    if 'parent_id' in vals and new_parent:
        # Update accounts in analytic lines before _sync_plan_column() unlinks child plan's column
        ...
    super().write(vals)
    if 'parent_id' in vals and not new_parent:
        # Update accounts in analytic lines after _sync_plan_column() creates the new column
        ...
```